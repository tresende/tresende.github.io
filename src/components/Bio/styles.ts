import styled from 'styled-components'

export const Wrapper = styled.p`
  line-height: 2rem;
`

export const SkillContainer = styled.ul`
  list-style: none;
`

export const Skill = styled.li`
  ::before {
    content: '-';
    padding-right: 0.5rem;
  }
`
