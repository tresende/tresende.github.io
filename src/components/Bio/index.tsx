import * as S from './styles'

const Bio = () => (
  <S.Wrapper>
    <p>Hello, friend.</p>
    <p>
      My name is Thiago Resende. I&apos;m a frontend developer living in Belo
      Horizonte, Brazil. I&apos;m a technology fan 💾, programming 👨‍💻, and
      coffee ☕️. These are my high-level skills:
    </p>
    <S.SkillContainer>
      <S.Skill>React</S.Skill>
      <S.Skill>Angular</S.Skill>
      <S.Skill>Node</S.Skill>
      <S.Skill>Flutter</S.Skill>
      <S.Skill>SNES</S.Skill>
      <S.Skill>Bass</S.Skill>
    </S.SkillContainer>
  </S.Wrapper>
)

export default Bio
