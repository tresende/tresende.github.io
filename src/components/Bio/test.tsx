import { render, screen } from '@testing-library/react'

import Bio from '.'

describe('<Bio />', () => {
  it('should render bio with skills', () => {
    const skils = ['React', 'Angular', 'Node', 'Flutter', 'SNES', 'Bass']

    const { container } = render(<Bio />)
    skils.forEach((skill) => expect(screen.getByText(skill)).toBeInTheDocument())

    expect(container).toMatchSnapshot()
  })
})
