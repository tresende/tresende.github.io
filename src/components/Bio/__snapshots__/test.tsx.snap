// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Bio /> should render bio with skills 1`] = `
.c0 {
  line-height: 2rem;
}

.c1 {
  list-style: none;
}

.c2::before {
  content: '-';
  padding-right: 0.5rem;
}

<div
  class="c0"
>
  <p>
    Hello, friend.
  </p>
  <p>
    My name is Thiago Resende. I'm a software engineer living in Belo Horizonte, Brazil. I'm a technology fan, programming and coffee. These are my high-level skills:
  </p>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      React
    </li>
    <li
      class="c2"
    >
      Angular
    </li>
    <li
      class="c2"
    >
      Node
    </li>
    <li
      class="c2"
    >
      Flutter
    </li>
    <li
      class="c2"
    >
      SNES
    </li>
    <li
      class="c2"
    >
      Bass
    </li>
  </ul>
</div>
`;
